Sources/WhisperKit/Core/Utils.swift

```diff
@@ -25,6 +25,14 @@ extension MLMultiArray {
         }
         return linearOffset
     }
+
+    func fill<Value>(indexes: [[NSNumber]], with value: Value) {
+        let pointer = UnsafeMutablePointer<Value>(OpaquePointer(dataPointer))
+        for index in indexes {
```

Comment:
This could take a while for large numbers of indexes - since this is a vector, it would be faster to parallelize this loop, here's one way to do it but open to other options (you could also use a taskgroup or similar) 
```suggestion
        DispatchQueue.concurrentPerform(iterations: indexes.count) { i in
            let index = indexes[i]
```

Reply:
Reference to the task group option https://stackoverflow.com/a/72540228

Reply:
created a small benchmark here https://github.com/jkrukowski/FillBenchmark to compare `fill` and `fill` with `DispatchQueue.concurrentPerform`. Looks like there is no speed up for a release build when using the latter.

Implementing `fill` using task group would most likely require changing the interface (`filterLogits` would have to be `async`) + it still might not bring the benefits, see the discussion [here](https://forums.swift.org/t/taskgroup-and-parallelism/51039/4). Would love to hear your opinion about it

Reply:
thanks! Merged your PR and updated readme with the new results in https://github.com/jkrukowski/FillBenchmark
Updated this PR with the precomputed version of the `linearOffset` function and added `@inline(__always)` directive which makes it equally performant as a standalone version. Keeping a separate `linearOffset` function might be useful in future

Reply:
`@inline(__always)` is neat! Agree it'll be useful to have around.